tests/hmm/hmumu_utils.py

```diff
@@ -2302,7 +2408,8 @@ def deltaphi_cudakernel(phi1, phi2, out_dphi):
 def get_theoryweights_cpu(offsets, variations, index, out_var):
     #loop over events
     for iev in numba.prange(len(offsets) - 1):
-        out_var[iev] = variations[offsets[iev]+index]
+        if(offsets[iev]+index < len(variations)):
```

Comment:
In case this `if` is False, would it be good to throw an error? Or can this happen intentionally, too?

Reply:
I had added this line when we didn't have the libiomp5 fix .. The `False` should never happen as such, I can add an exception error.

Reply:
When you access an array out of bounds in numba/C/C++, generally the program will not crash or give an error, but silently produce wrong data.

I would recommend to add `NUMBA_BOUNDSCHECK=1` to `hmm_analyze.sh` and `run.sh` to automatically crash on such cases, then you don't need to do an explicit check every time. You will then need to make sure that the variations you ask for correspond to what is available. Here I'm worried that you just truncate without any warning, so in the future this might be a spot for errors if things change.

The segfault fix in PR#90 was related, but the underlying issues were different: the underlying problem there was defining an array as `int8` in one place but using it as `int32` in another place, causing memory issues.

Reply:
Ok, done. Thanks for the suggestion!